controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -0,0 +1,144 @@
+        return 1;
+    }
+    catch (const YAML::ParserException& e) {
+        std::cerr << e.msg << std::endl;
+        return 2;
+    }
+
+    webots::Node* tempNode = supervisor.getFromDef("test");
+    xSize                  = tempNode->getField("xSize")->getSFFloat();
+    ySize                  = tempNode->getField("ySize")->getSFFloat();
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+
+    // Generate distributions for random number systems in the following loop
+    std::uniform_int_distribution<> xDistrib(0, xSize * 100);
+    std::uniform_int_distribution<> yDistrib(0, ySize * 100);
+    std::uniform_int_distribution<> rotDistrib(0, rotations.size() - 1);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Declare a vector of positions that will be saved as they are randomly generated, to be later
+        // applied to each robot
+        std::vector<std::array<double, 3>> positions = std::vector<std::array<double, 3>>();
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Assume there is no collision
+            bool collision = false;
+            // newPos will be a "Proposed location" for a robot to teleport to
+            std::array<double, 3> newPos;
+            do {
+                collision = false;
+                // Generate a new random location
+                newPos[0] = xSize / 2 - xDistrib(gen) / 100;
+                newPos[1] = ySize / 2 - yDistrib(gen) / 100;
+                newPos[2] = zHeight;
+
+                // Loop through the vector of existing proposed locations and see if the new one is going to
+                // collide with any of them
+                for (std::array<double, 3> testPos : positions) {
+                    double distance = sqrt(pow((newPos[1] - testPos[1]), 2) + pow((newPos[0] - testPos[0]), 2));
+                    if (distance < minDistance) {
+                        collision = true;
+                    }
+                }
+            }
+            // Loop until a proposed location has been found that doesn't clash with the existing ones
+            while (collision);
+            // Finally add the proposed location in as a confirmed position
+            positions.push_back(newPos);
+        }
+
+        // Loop through every robot
+        for (size_t i = 0; i < otherRobotsNodes.size(); i++) {
+            // Grab translation field of the robot to modify
+            webots::Field& target_translation_field = *(otherRobotsNodes[i])->getField("translation");
+            // Grab the current rotation field of the robot to modify
+            webots::Field& target_rotation_field = *(otherRobotsNodes[i])->getField("rotation");
```

Comment:
Just leave these as pointers. There is no real need/benefit to dereferencing them
```suggestion
            webots::Field* target_translation_field = otherRobotsNodes[i]->getField("translation");
            // Grab the current rotation field of the robot to modify
            webots::Field* target_rotation_field = otherRobotsNodes[i]->getField("rotation");
```

Reply:
I mean, I do this all the time so this one is a matter of opinion :P

It means I don't have to use `->` later!

Reply:
I originally changed them from pointers I think because we were trying to avoid using raw pointers as much as possible and I couldn't get smart pointers to work with webots... But I don't mind either way :)